reports/ri_ec2.go

```diff
@@ -0,0 +1,131 @@
+//   Copyright 2018 MSolution.IO
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package reports
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/trackit/jsonlog"
+
+	"github.com/trackit/trackit-server/aws"
+	"github.com/trackit/trackit-server/aws/usageReports/history"
+	"github.com/trackit/trackit-server/usageReports/riEc2"
+	"github.com/trackit/trackit-server/users"
+)
+
+var riEc2InstanceFormat = [][]cell{{
+	newCell("", 7).addStyle(textCenter, textBold, backgroundGrey),
+	newCell("Offering", 2).addStyle(textCenter, textBold, backgroundGrey),
+	newCell("Recurring Charges", 2).addStyle(textCenter, textBold, backgroundGrey),
+	newCell("Reservation", 2).addStyle(textCenter, textBold, backgroundGrey),
+	newCell("", 1).addStyle(textCenter, textBold, backgroundGrey),
+}, {
+	newCell("Account").addStyle(textCenter, textBold, backgroundGrey),
+	newCell("ID").addStyle(textCenter, textBold, backgroundGrey),
+	newCell("Type").addStyle(textCenter, textBold, backgroundGrey),
+	newCell("Region").addStyle(textCenter, textBold, backgroundGrey),
+	newCell("Description").addStyle(textCenter, textBold, backgroundGrey),
+	newCell("Count").addStyle(textCenter, textBold, backgroundGrey),
+	newCell("Tenancy").addStyle(textCenter, textBold, backgroundGrey),
+	newCell("Class").addStyle(textCenter, textBold, backgroundGrey),
+	newCell("Type").addStyle(textCenter, textBold, backgroundGrey),
+	newCell("Amount").addStyle(textCenter, textBold, backgroundGrey),
+	newCell("Frequency").addStyle(textCenter, textBold, backgroundGrey),
+	newCell("Start").addStyle(textCenter, textBold, backgroundGrey),
+	newCell("End").addStyle(textCenter, textBold, backgroundGrey),
+	newCell("Tags").addStyle(textCenter, textBold, backgroundGrey),
+}}
+
+func formatRiEc2Instance(report riEc2.ReservationReport) []cell {
+	reservation := report.Reservation
+	tags := formatTags(reservation.Tags)
+	chargesAmount := make([]string, 0)
+	chargesFrequency := make([]string, 0)
```

Comment:
You can use `len(reservation.RecurringCharges)` instead of 0. So that, it won't allocate again and again memory.